### ISSUE TITLE
[Example] image super-resolution

### ISSUE THREAD

**Issue by GreyZzzzzzXh** [intel/webml-polyfill#253]

**Comment by huningxin** [comment]
@GreyZzzzzzXh , [SRGAN](https://github.com/tensorlayer/srgan) might be interested by you.

**Comment by GreyZzzzzzXh** [comment]
1. I trained the [SRGAN](https://arxiv.org/abs/1609.04802) model on DIV2K dataset using tensorlayer. 
Reference: [tensorlayer/srgan](https://github.com/tensorlayer/srgan)

2. The generator of SRGAN is called SRResNet, being like this. 
![screenshot from 2018-12-12 12-27-12](https://user-images.githubusercontent.com/35633199/49847284-4d9c0a00-fe0a-11e8-9966-dcfaad82b5ef.png)
We can achieve a 4x upscaling.
According to the origin paper, set B=16.

3. SRResNet uses operation PixelShuffler([DepthToSpace](https://www.tensorflow.org/api_docs/python/tf/nn/depth_to_space))  which is not supported in tflite yet, see [TfLite buildin ops](https://github.com/tensorflow/tensorflow/blob/454fd39652993f387bafb83329e93cce7c0db9db/tensorflow/lite/builtin_ops.h). So i use **transpose + batchToSpace + transpose** instead. 
New ops we need: [transpose](https://www.tensorflow.org/api_docs/python/tf/transpose), [batchToSpace](https://www.tensorflow.org/api_docs/python/tf/batch_to_space), [tanh](https://www.tensorflow.org/api_docs/python/tf/math/tanh)

4. An easy image demo with WebGL backend. 
input size [96, 96, 3] => output size [384, 384, 3]
It takes several seconds to refer now.
![screenshot from 2018-12-12 13-56-26](https://user-images.githubusercontent.com/35633199/49849730-e71ce900-fe15-11e8-900d-bbc4c32198d1.png)
Visit https://greyzzzzzzxh.github.io/webml-examples to test.

**Comment by huningxin** [comment]
Very nice! We need to investigate the platform API support for the three new ops. BTW, the [link](https://greyzzzzzzxh.github.io/webml-examples/) doesn't work for me. Would you mind to double check it?

**Comment by GreyZzzzzzXh** [comment]
> the link doesn't work

updated it.

**Comment by huningxin** [comment]
The link works, however it reports WebGL backend failure to me. Did I miss any thing?

**Comment by GreyZzzzzzXh** [comment]
Is there any error or warning? I can get results on linux and android via this link.

**Comment by huningxin** [comment]
The failure disappears now. It's super (for super resolution). 👍

**Comment by huningxin** [comment]
OpenVINO pre-trained models also contains a SR model (https://software.intel.com/en-us/openvino-toolkit/documentation/pretrained-models). Please take a look.

**Comment by GreyZzzzzzXh** [comment]
![screenshot from 2018-12-29 16-37-08](https://user-images.githubusercontent.com/35633199/50535742-13f82c80-0b88-11e9-9572-9fe52000967a.png)

I trained FSRCNN, a small network for super resolution.

New ops we need: [sigmoid](https://www.tensorflow.org/api_docs/python/tf/math/sigmoid), [maximum](https://www.tensorflow.org/api_docs/python/tf/math/maximum), [tanh](https://www.tensorflow.org/api_docs/python/tf/math/tanh), [transpose_conv](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d_transpose)

Simple image demo with WebGL backend:

![screenshot from 2018-12-29 16-44-12](https://user-images.githubusercontent.com/35633199/50535843-114a0700-0b89-11e9-8270-732b5bb86f9c.png)


Reference: [Accelerating the Super-Resolution Convolutional Neural Network](http://mmlab.ie.cuhk.edu.hk/projects/FSRCNN.html)

Visit https://greyzzzzzzxh.github.io/webml-examples/index.html to test.

**Comment by ibelem** [comment]
Move to Feb 2019 milestone for this issue.
